### PR TITLE
Brings Justice to the Station

### PR DESCRIPTION
**Pull request by TheDZD**
Port of tgstation/-tg-station#8519 with tweaked sounds from tgstation/-tg-station#10496, and a few other tweaks. Also a bit of clothing variable refactoring from -tg-, some of which allows for helmets with toggle-able states like visors and such.

Has this literal cancer caused the quality of Github PRs to hit rock bottom?
Has science gone too far?
Have I lost my mind?
The answer to all of those questions is, without a doubt, **absolutely**.

:cl:
add: Adds justice helmets with flashing lights and annoying sirens.
add: Adds crates containing said helmets.
add: Also includes a version of the helmet that can only be spawned by admins, and just has a single red light in the center instead of the stereotypical police blinkers.
/:cl:

### CODE REVIEW

**Comment by SirBloodBuske** [comment]
Dear god, ITS FINNALLY HERE!

**Comment by ZomgPonies** [comment]
OMG THIS PR WILL ENCOURAGE FURRIES AND HELMET FETISHISTS

File: code/modules/clothing/head/helmet.dm [hunk 1]
```diff
@@ -16,6 +16,24 @@
...
+			flags ^= visor_flags
+			flags_inv ^= visor_flags_inv
+			icon_state = "[initial(icon_state)][up ? "up" : ""]"
+			user << "[up ? alt_toggle_message : toggle_message] \the [src]"
+
+			user.update_inv_head()
+
+			if(active_sound)
+				while(up)
+					playsound(src.loc, "[active_sound]", 100, 0, 4)
```
Comment (Tastyfish) [inline code/modules/clothing/head/helmet.dm]: 100 for a clear, continuous noise? rip ears

**Comment by TullyBurnalot** [comment]
Needs more echo and siren.

Nevah enuff siren.

File: code/modules/clothing/head/helmet.dm [hunk 2]
```diff
@@ -66,6 +84,25 @@
 	flags_inv = HIDEEARS
 	strip_delay = 80
 
+/obj/item/clothing/head/helmet/justice
+	name = "helmet of justice"
+	desc = "WEEEEOOO. WEEEEEOOO. WEEEEOOOO."
+	icon_state = "justice"
+	toggle_message = "You turn off the lights on"
```
Comment (monster860) [inline code/modules/clothing/head/helmet.dm]: Why the extra "on" at the end

File: code/modules/clothing/head/helmet.dm [hunk 3]
```diff
@@ -66,6 +84,25 @@
 	flags_inv = HIDEEARS
 	strip_delay = 80
 
+/obj/item/clothing/head/helmet/justice
+	name = "helmet of justice"
+	desc = "WEEEEOOO. WEEEEEOOO. WEEEEOOOO."
+	icon_state = "justice"
+	toggle_message = "You turn off the lights on"
+	alt_toggle_message = "You turn on the lights on"
```
Comment (monster860) [inline code/modules/clothing/head/helmet.dm]: Why the extra "on" at the end

File: code/modules/clothing/head/helmet.dm (hunk 2 above)
Comment (TheDZD) [inline code/modules/clothing/head/helmet.dm]: End result for this is: "You turn off the lights on the helmet of justice."

File: code/modules/clothing/head/helmet.dm (hunk 3 above)
Comment (TheDZD) [inline code/modules/clothing/head/helmet.dm]: See above.

**Comment by NTSAM** [comment]
:-1: because it's fun


(I'm kidding :+1: because it's fun i'm not a douche)

**Comment by KasparoVy** [comment]
crank the 'fuck yes' dial up past 11

**Comment by CrAzYPiLoT-SS13** [comment]
IPC Robocop when? :+1:

**Comment by FalseIncarnate** [comment]
Just waiting for someone to slap these onto a bunch of sentient monkeys and let them loose.

:+1:

**Comment by monster860** [comment]
Oh god... No. Dont put them on sentient monkeys. Ever. Bad idea.

**Pull request merged**